addons/binding/org.openhab.binding.yamahareceiver/src/main/java/org/openhab/binding/yamahareceiver/internal/protocol/YamahaReceiverCommunication.java

```diff
@@ -293,4 +316,153 @@ private String postAndGetResponse(String message) throws IOException {
             }
         }
     }
+
+    private class NetRadioMenu {
```

Comment:
It's a private non-static class, because it is using communication helper methods like `postAndGetResponse`. And they are using instance properties from the outer class.
Extracting to separate file means extracting communication helper methods to a utils class first and change their signature, thus causing several changes in old code.

With that in mind, do you still prefer the separate file?